db/migrate/20150508124600_copy_unmanaged_hosts_to_interfaces.rb

```diff
@@ -19,9 +19,13 @@ def up
     say "Migrating Unmanaged Host interfaces to standalone Interfaces"
 
     FakeHost.where(:managed => false).each do |host|
+      say "  Migrating #{host.name}"
       nic = FakeNic.where(:host_id => host.id, :primary => true, :provision => true).first
       if nic.present?
         nic.name = host.name if nic.name.blank?
+        # we ignore validation errors on existing primary interface
+        say "Host #{host.name} has invalid interface #{nic.identifier}, you have to fix it manually through API" unless nic.valid?
```

Comment:
Unfortunately that's not entirely truth. We'll allow changing the identifier but if the interface is invalid because of IP or MAC, we can't fix that through UI because these validations are executed in host model and it operates on saved interfaces (the incoming parameters aren't applied yet, one of the downside of using nested attributes). Through API you can modify interface directly.

Otherwise I agree and will add more information to the message.